src/vs/workbench/contrib/tasks/electron-sandbox/taskService.ts

```diff
@@ -132,12 +132,19 @@ export class TaskService extends AbstractTaskService {
 			return this._taskSystem;
 		}
 		this._taskSystem = this._createTerminalTaskSystem();
-		this._taskSystemListener = this._taskSystem!.onDidStateChange((event) => {
+		this._taskSystemListeners = [this._taskSystem!.onDidStateChange((event) => {
 			if (this._taskSystem) {
 				this._taskRunningState.set(this._taskSystem.isActiveSync());
 			}
 			this._onDidStateChange.fire(event);
-		});
+		}),
+			this._taskSystem!.onDidReconnectToTerminals((event) => {
+				if (this._taskSystem) {
+					this._taskRunningState.set(this._taskSystem.isActiveSync());
+				}
+				this._onDidReconnectToTerminals.fire(event);
+			})
+		];
```

Comment:
You can avoid the `!` and if you do:

```
const taskSystem = this._createTerminalTaskSystem();
this._taskSystem = taskSystem;
```

The if statement should be fine to remove since it's a listener on this._taskSystem so it's guaranteed to exist if that event is disposed of correctly.

Reply:
would have to do that within the event listener as well to remove all `!` but should be safe